esmvaltool/cmorizers/data/formatters/datasets/esdc.py

```diff
@@ -0,0 +1,149 @@
+    20230126
+
+Download and processing instructions
+    It is not necessary to download the data, as the cmorizer script can access
+    it directly from the cloud if it is not available locally.
+
+    To download a dataset, the dataset folder can be explored on the source
+    website, and downloaded using wget:
+        ```wget -m -nH -np -R "index.html*" http://data.rsc4earth.de/EarthSystemDataCube/v3.0.1/```
+"""  # noqa: E501
+import logging
+from copy import deepcopy
+from pathlib import Path
+
+import cf_units
+import iris.std_names
+import xarray as xr
+from esmvalcore.preprocessor import monthly_statistics
+
+from esmvaltool.cmorizers.data import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def _fix_cube(var, cube, cfg):
+    """General fixes for all cubes."""
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], var['short_name'])
+
+    # Set correct names
+    cube.var_name = cmor_info.short_name
+    if cmor_info.standard_name:
+        cube.standard_name = cmor_info.standard_name
+    cube.long_name = cmor_info.long_name
+
+    # Set calendar to gregorian instead of proleptic gregorian
+    old_unit = cube.coord('time').units
+    if old_unit.calendar == 'proleptic_gregorian':
+        logger.info("Converting time units to gregorian")
+        new_unit = cf_units.Unit(old_unit.origin, calendar='gregorian')
+        cube.coord('time').units = new_unit
+    utils.fix_coords(cube)
+    cube.convert_units(cmor_info.units)
+    if 'height2m' in cmor_info.dimensions:
+        utils.add_height2m(cube)
+    # Conversion from 8-d to monthly frequency
+    cube = monthly_statistics(cube, operator="mean")
+
+    # Fix metadata
+    attrs = cfg['attributes']
+    attrs['mip'] = var['mip']
+    utils.fix_var_metadata(cube, cmor_info)
+    utils.set_global_atts(cube, attrs)
+
+    return cube
+
+
+def _open_zarr(path):
+    """Open zarr dataset."""
+    logger.info('Opening zarr in "%s"', path)
+    try:
+        zarr_dataset = xr.open_dataset(path, engine='zarr')
+        return zarr_dataset
+    except KeyError as exception:
+        # Happens when the zarr folder is missing metadata, e.g. when
+        # it is a zarr array instead of a zarr dataset.
+        logger.error('Could not open zarr dataset "%s": "KeyError: %s"', path,
+                     exception)
+        raise exception
+
+
+def _extract_variable(zarr_path, var, cfg, out_dir):
+    """Open and cmorize cube."""
+    attributes = deepcopy(cfg['attributes'])
+    all_attributes = {
+        **attributes,
+        **var
+    }  # add the mip to the other attributes
+    raw_name = var['raw']
+    zarr_dataset = _open_zarr(zarr_path)
+    cube_xr = zarr_dataset[raw_name]
+
+    # Invalid standard names must be removed before converting to iris
+    standard_name = cube_xr.attrs.get('standard_name', None)
+    if (standard_name is not None
+            and standard_name not in iris.std_names.STD_NAMES):
+        del cube_xr.attrs['standard_name']
```

Comment:
Good question, I should have asked for recommendations on how to handle that.

The problem I was encountering is that the `standard_name` provided by the dataset is not recognized by Iris, so it crashed when doing `to_iris()`. I deleted it because later, following other cmorizers, the standard name is overriden/set again (https://github.com/ESMValGroup/ESMValTool/blob/4877463d5c59140c071be366fefae818ffcb3df7/esmvaltool/cmorizers/data/formatters/datasets/esdc.py#L40-L41).

If you think it's better to just override the value instead of deleting the pair we can do that, I have tested it and it works fine as well.

Reply:
Speaking of this code, the check is a bit crude for my taste. The actual test in iris is a bit more complex. Unfortunately, the code is not easily reusable. Can we keep it or should we do something about it?

https://github.com/SciTools/iris/blob/9268ca9c70bd160c35c5099025d632ae33faf858/lib/iris/common/mixin.py#L23-L52

Reply:
well let's go along the "don't fix it if ain't broken" - if this works fine then it's OK to leave it as is :+1: